test/foundry/smart-account/SA.Basics.t.sol

```diff
@@ -49,4 +62,63 @@ contract SABasicsTest is SATestBase {
             "smart account should have 1 token"
         );
     }
+
+    function testExecuteBatch() external {
+        uint256 smartAccountDeploymentIndex = 0;
+        bytes memory moduleSetupData = getEcdsaOwnershipRegistryModuleSetupData(
+            alice.addr
+        );
+        SmartAccount sa = getSmartAccountWithModule(
+            address(ecdsaOwnershipRegistryModule),
+            moduleSetupData,
+            smartAccountDeploymentIndex,
+            "aliceSA"
+        );
+
+        address[] memory dest = new address[](2);
+        uint256[] memory values = new uint256[](2);
+        bytes[] memory calldatas = new bytes[](2);
+
+        vm.expectEmit(true, true, true, true);
+        emit Log("hello");
+        vm.expectEmit(true, true, true, true);
+        emit Log("world");
+
+        dest[0] = address(test);
+        dest[1] = address(test);
+        calldatas[0] = abi.encodeCall(test.emitString, ("hello"));
+        calldatas[1] = abi.encodeCall(test.emitString, ("world"));
+
+        UserOperation memory op = makeEcdsaModuleUserOp(
+            getSmartAccountBatchExecuteCalldata(dest, values, calldatas),
+            sa,
+            0,
+            alice
+        );
+        vm.breakpoint("a");
+        entryPoint.handleOps(arraifyOps(op), owner.addr);
+    }
+
+    function testReceiveEtherWithGasLimit() external {
+        // Deploy Smart Account with default module
+        uint256 smartAccountDeploymentIndex = 0;
+        bytes memory moduleSetupData = getEcdsaOwnershipRegistryModuleSetupData(
+            alice.addr
+        );
+        SmartAccount sa = getSmartAccountWithModule(
+            address(ecdsaOwnershipRegistryModule),
+            moduleSetupData,
+            smartAccountDeploymentIndex,
+            "aliceSA"
+        );
+
+        MockEthSender mockEthSender = new MockEthSender();
+        vm.deal(address(mockEthSender), 100 ether);
+        
+        uint256 gasStipend = 0;
+        // by some reason it passes even if there's extra gas consumed by receive()
```

Comment:
I think you wanted to chat about this? we could have a call then I will also check it out on local